From c146122d2f3c647611e8862470b28b504b4d71da Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Wed, 10 Aug 2016 10:21:48 -0400
Subject: [PATCH 646/723] HIVE-14457: Partitions in encryption zone are still
 trashed though an exception is returned (Chaoyu
 Tang, reviewed by Yongzhi Chen, Sergio Pena)

Change-Id: Ic12f19082f5aaab409842ce40382ada33716e8d1
(cherry picked from commit 435443c3f90c1eacc8598eeb2e471c944975e659)
---
 .../hadoop/hive/metastore/HiveMetaStore.java       |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
index 3f5c6cb..65a7555 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
@@ -2718,16 +2718,18 @@ private boolean drop_partition_common(RawStore ms, String db_name, String tbl_na
           checkTrashPurgeCombination(archiveParentDir, db_name + "." + tbl_name + "." + part_vals,
               mustPurge, deleteData && !isExternalTbl);
         }
-        if (!ms.dropPartition(db_name, tbl_name, part_vals)) {
-          throw new MetaException("Unable to drop partition");
-        }
-        success = ms.commitTransaction();
+
         if ((part.getSd() != null) && (part.getSd().getLocation() != null)) {
           partPath = new Path(part.getSd().getLocation());
           verifyIsWritablePath(partPath);
           checkTrashPurgeCombination(partPath, db_name + "." + tbl_name + "." + part_vals,
               mustPurge, deleteData && !isExternalTbl);
         }
+
+        if (!ms.dropPartition(db_name, tbl_name, part_vals)) {
+          throw new MetaException("Unable to drop partition");
+        }
+        success = ms.commitTransaction();
       } finally {
         if (!success) {
           ms.rollbackTransaction();
-- 
1.7.9.5

